crates/pipeline_manager/src/db/mod.rs

```diff
@@ -261,15 +262,170 @@ pub(crate) struct ProgramDescr {
     pub schema: Option<ProgramSchema>,
 }
 
-/// Lifecycle of a pipeline.
-#[derive(Serialize, ToSchema, Eq, PartialEq, Debug, Clone, Copy)]
+/// Pipeline status.
+///
+/// This type represents the state of the pipeline tracked by the pipeline runner and
+/// observed by the API client via the `GET /pipeline` endpoint.
+///
+/// ### The lifecycle of a pipeline
+///
+/// The following automaton captures the lifecycle of the pipeline.  Individual states
+/// and transitions of the automaton are described below.
+///
+/// * In addition to the transitions shown in the diagram, all states have an implicit
+///   "forced shutdown" transition to the `Shutdown` state.  This transition is triggered
+///   when the pipeline runner is unable to communicate with the pipeline or and forces
```

Comment:
"This transition is triggered when the pipeline runner is unable to communicate with the pipeline and thereby forces a shutdown. How the shutdown is forced will depend on the type of Runner; e.g., the local runner might send the pipeline a `KILL` signal."